app/controllers/user_plans_controller.rb

```diff
@@ -1,2 +1,20 @@
 class UserPlansController < ApplicationController
+  def new
+    @user_plan = UserPlan.new
+  end
+
+  def create
+    @user_plan = UserPlan.create(user_plan_params)
+    if @user_plan.save
+      redirect_to personal_dashboard_path
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def user_plan_params
+    params.require(:user_plan).permit(:order_id, :start_date, :end_date)
```

Comment:
Does the user pass the order_id, start_date, and end_date? I don't think this should go in the user_plan_params. Rather define these sings in the create method.